examples/channel_routing.rs

```diff
@@ -0,0 +1,43 @@
+//! Channel router example
+
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    use rodio::source::{Function, SignalGenerator, Source};
+    use std::thread;
+    use std::time::Duration;
+
+    let stream_handle = rodio::OutputStreamBuilder::open_default_stream()?;
+
+    // let test_signal_duration = Duration::from_millis(1000);
+    let interval_duration = Duration::from_millis(100);
+    let sample_rate = cpal::SampleRate(48000);
+
+    let (mut controller, router) = SignalGenerator::new(sample_rate, 1000.0, Function::Triangle)
+        .amplify(0.1)
+        .channel_router(2, vec![vec![0.0f32, 0.0f32]]);
+
+    println!("Playing 1000Hz tone");
+
+    stream_handle.mixer().add(router);
+
+    for i in 0..1000 {
+        thread::sleep(interval_duration);
+        let n = i % 20;
+        match n {
+            0 => println!("Left speaker ramp up"),
+            1..10 => {
+                _ = controller.map(0, 0, n as f32 / 10.0);
+                _ = controller.map(0, 1, 0f32);
```

Comment:
this is not very readable right? I can guess its channel from, channel too, gain? But we might be able to do better. I see a number of options, there might well be more:
1️⃣ make map take a struct like: `controller.map(Mapping { from: 0, to: 1, gain: 0f32 })`

2️⃣ make map a builder: `controller.map().from(0).to(1).with_gain(0.f32).appy()`. Here you could make the gain optional and do a default of 1.0

3️⃣ rename it so the argument names are in the name: `controller.map_from_to_with_gain(0, 1, 0f32)`.

I think 2 is my favorite, though its a little more code.

Reply:
I like the builder too.